tools/reviewer/job/sync/Patch.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.reviewer.job.sync;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Parses `patch` of GitHub Pull Request file. `Patch` of GitHub Pull Request file can involve one
+ * or more parts.
+ */
+public class Patch {
+  /**
+   * The regex pattern for all newline symbols search: `\\n\\s` - newline with space, `\\n\\+` -
+   * newline with `+` character, `\\n\\-` - newline with `-` character, `\\n` - just newline
+   */
+  private static final String NEW_LINES_PATTERN = "\\n\\s|\\n\\+|\\n\\-|\\n";
+  /** The regex pattern for DiffHunkHeader headers search. */
```

Comment:
Can you give an example of a diff hunk header? Or mention the explanation below.